staging/provisioning/windows/hostsconfigagent.ps1

```diff
@@ -0,0 +1,54 @@
+$Global:ClusterConfiguration = ConvertFrom-Json ((Get-Content "c:\k\kubeclusterconfig.json" -ErrorAction Stop) | out-string)
+$clusterFQDN = $Global:ClusterConfiguration.Kubernetes.ControlPlane.IpAddress
+$hostsFile="C:\Windows\System32\drivers\etc\hosts"
+$retryDelaySeconds = 15
+
+# Dot-source scripts with functions that are called in this script
+. c:\AzureData\k8s\kuberneteswindowsfunctions.ps1
+
+function Get-APIServer-IPAddress
+{
+    $uri = "http://169.254.169.254/metadata/instance/compute/tags?api-version=2019-03-11&format=text"
+    $response = Retry-Command -Command "Invoke-RestMethod" -Args @{Uri=$uri; Method="Get"; ContentType="application/json"; Headers=@{"Metadata"="true"}} -Retries 3 -RetryDelaySeconds 5
```

Comment:
How often does the ip address change? can this be cached so there are not continuous calls?

Reply:
it depends. But when it changed, the script would catch it as soon as possible. So I think depending on intervals should be better. Do you insist on the caching, or do you think we should increase the interval?

Reply:
I don't have enough context. It looks like the [linux solution](https://github.com/Azure/aks-engine/pull/3556/files) is doing the same so I say we keep similar